### PR TITLE
[GH-#199] Better function specs, general cleanup

### LINKED ISSUE #199: Better function specs, general cleanup

### PR DESCRIPTION
**Pull request by d-velev**
closes #199

### CODE REVIEW

File: apps/aecore/lib/aecore/chain/block_validation.ex [hunk 1]
```diff
@@ -45,7 +45,7 @@ defmodule Aecore.Chain.BlockValidation do
     end
   end
 
-  @spec single_validate_block!(Block.t()) :: {:error, term()} | :ok
+  @spec single_validate_block!(Block.t) :: {:error, term()} | :ok
```
Comment (untra) [inline apps/aecore/lib/aecore/chain/block_validation.ex]: So are you keeping the erlang-style parens on the types or not? 😩 
Some types have parens, some don't. I would greatly encourage consistency here, and sticking with it.

Comment (thepiwo) [reply apps/aecore/lib/aecore/chain/block_validation.ex]: @d-velev @cytadela8 havent we decided to keep the erlang-style parens?

File: apps/aehttpclient/lib/client.ex [hunk 2]
```diff
@@ -90,14 +89,14 @@ defmodule Aehttpclient.Client do
     get(uri <> "/tx_pool/#{acc}", :acc_txs)
   end
 
-  @spec handleResponse(:block, map(), map()) :: {:ok, map()}
-  defp handleResponse(:block, body, _headers) do
+  @spec handle_response(:block, map(), map()) :: {:ok, map()}
```
Comment (untra) [inline apps/aehttpclient/lib/client.ex]: `@spec handle_response(req_kind(), map(), list(map())) :: {:ok, map()}`
This spec can be defined once at the top, and then doesn't need to be repeated for each atom defined in `req_kind`

File: apps/aecore/lib/aecore/chain/block_validation.ex [hunk 3]
```diff
@@ -8,11 +8,11 @@ defmodule Aecore.Chain.BlockValidation do
   alias Aecore.Chain.ChainState
   alias Aecore.Chain.Difficulty
 
-  @spec calculate_and_validate_block!(Block.t(), Block.t(), map(), list(Block.t())) :: {:error, term()} | :ok
+  @spec calculate_and_validate_block!(Block.t(), Block.t(), Chainstate.account_chainstate, list(Block.t())) :: {:error, term()} | :ok
```
Comment (cytadela8) [inline apps/aecore/lib/aecore/chain/block_validation.ex]: Parenthesis after `Chainstate.account_chainstate`

**Comment by thepiwo** [comment]
@meivantodorov can you review?

**Comment by cytadela8** [comment]
@d-velev Still a lot of missing parenthesis. (including the example)

**Pull request merged**